projects/kubernetes/autoscaler/1-23/patches/0002-Fix-CVE-2022-1996-for-1.23.patch

```diff
@@ -0,0 +1,50 @@
+From 8fece6513f3a3943d3116515cb15bed6ba2b675d Mon Sep 17 00:00:00 2001
+From: Prow Bot <prow@amazonaws.com>
+Date: Wed, 8 Feb 2023 10:18:23 -0500
+Subject: [PATCH] Fix CVE 2022-1996 for 1.23
+
+Requires go-restful>=3.8.0.
+---
+ cluster-autoscaler/go.mod | 3 +++
+ cluster-autoscaler/go.sum | 7 ++++---
+ 2 files changed, 7 insertions(+), 3 deletions(-)
+
+diff --git a/cluster-autoscaler/go.mod b/cluster-autoscaler/go.mod
+index 2bed8b24d..80f97ed21 100644
+--- a/cluster-autoscaler/go.mod
++++ b/cluster-autoscaler/go.mod
+@@ -12,6 +12,7 @@ require (
+ 	github.com/Azure/go-autorest/autorest/to v0.4.0
+ 	github.com/aws/aws-sdk-go v1.38.49
+ 	github.com/digitalocean/godo v1.27.0
++	github.com/emicklei/go-restful/v3 v3.10.0 // indirect
+ 	github.com/ghodss/yaml v1.0.0
+ 	github.com/gofrs/uuid v4.0.0+incompatible
+ 	github.com/golang/mock v1.6.0
+@@ -101,3 +102,5 @@ replace k8s.io/sample-cli-plugin => k8s.io/sample-cli-plugin v0.23.0
+ replace k8s.io/sample-controller => k8s.io/sample-controller v0.23.0
+ 
+ replace k8s.io/pod-security-admission => k8s.io/pod-security-admission v0.23.0
++
++replace github.com/emicklei/go-restful => github.com/emicklei/go-restful/v3 v3.10.1
+diff --git a/cluster-autoscaler/go.sum b/cluster-autoscaler/go.sum
+index 046b77812..9b4114772 100644
+--- a/cluster-autoscaler/go.sum
++++ b/cluster-autoscaler/go.sum
+@@ -235,9 +235,10 @@ github.com/dustin/go-humanize v1.0.0 h1:VSnTsYCnlFHaM2/igO1h6X3HA71jcobQuxemgkq4
+ github.com/dustin/go-humanize v1.0.0/go.mod h1:HtrtbFcZ19U5GC7JDqmcUSB87Iq5E25KnS6fMYU6eOk=
+ github.com/elazarl/goproxy v0.0.0-20180725130230-947c36da3153 h1:yUdfgN0XgIJw7foRItutHYUIhlcKzcSf5vDpdhQAKTc=
+ github.com/elazarl/goproxy v0.0.0-20180725130230-947c36da3153/go.mod h1:/Zj4wYkgs4iZTTu3o/KG3Itv/qCCa8VVMlb3i9OVuzc=
+-github.com/emicklei/go-restful v0.0.0-20170410110728-ff4f55a20633/go.mod h1:otzb+WCGbkyDHkqmQmT5YD2WR4BBwUdeQoFo8l/7tVs=
+-github.com/emicklei/go-restful v2.9.5+incompatible h1:spTtZBk5DYEvbxMVutUuTyh1Ao2r4iyvLdACqsl/Ljk=
+-github.com/emicklei/go-restful v2.9.5+incompatible/go.mod h1:otzb+WCGbkyDHkqmQmT5YD2WR4BBwUdeQoFo8l/7tVs=
++github.com/emicklei/go-restful/v3 v3.10.0 h1:X4gma4HM7hFm6WMeAsTfqA0GOfdNoCzBIkHGoRLGXuM=
```

Comment:
going from a 2.x to 3.x seems risky. Did we do any validation here to ensure 3.x is still compatible with whatever dep is pulling it in?

Reply:
I did a few things to verify:

1. The [changelog](https://github.com/emicklei/go-restful/blob/v3/CHANGES.md#v310) does not indicate any breaking changes. It seems like they used the v3 rev to cut over to go modules.
2. I checked the [diff](https://github.com/emicklei/go-restful/compare/v3.0.0..jonathanmeier5:go-restful:v2.16.0) and did not find any backward incompatible changes.
3. The build succeeds, indicating to me that the public API did not include backward incompatible changes from v2.x to v3.x.
4. The built container properly scales machine deployments up and down when tested manually on a docker cluster.

Happy to do more testing if we feel it is necessary.